volume/store/store.go

```diff
@@ -269,14 +273,16 @@ func (s *VolumeStore) create(name, driverName string, opts, labels map[string]st
 		if v.DriverName() != driverName && driverName != "" && driverName != volume.DefaultDriverName {
 			return nil, errNameConflict
 		}
-		return v, nil
+		// In case a volume has already been created, we will return an error (non-idempotency)
+		return v, errAlreadyCreated
```

Comment:
I'm not a huge fan of this approach.

Ultimately the reason the code is as it is was to support the legacy auto-create/use for `docker run -v <name>:/foo`.
Meanwhile this pattern is racey:

```
Get(foo) || Create(foo)
```

Maybe we could have `Get` accept a callback that gets run at the end of the fn while the store is still locked.